pkg/kubectl/cmd/expose.go

```diff
@@ -149,9 +149,9 @@ func RunExpose(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []str
 	// For objects that need a pod selector, derive it from the exposed object in case a user
 	// didn't explicitly specify one via --selector
 	if s, found := params["selector"]; found && kubectl.IsZero(s) {
-		s, err := f.PodSelectorForObject(inputObject)
```

Comment:
Will expose support replicasets? If so, we need to add replicasets to `CanBeExposed` in factory.go. And modify the description of `kubectl expose` in this file too. This can be done in a follow up PR.